cpp/src/slice2cs/CsUtil.cpp

```diff
@@ -51,6 +51,12 @@ Slice::interfaceName(const ClassDefPtr& c)
     return name.find("II") == 0 ? name : "I" + name;
 }
 
+std::string
+Slice::structName(const StructPtr& s)
+{
+    return normalizeCase(s) ? pascalCase(s->name()) : s->name();
```

Comment:
normalizeCase does not affect the identifiers for types (class, struct, interface, exception, enum). They are always kept as-is, which is usually PascalCase. ReturnValue structs and similar - that Ice creates from non-type identifiers - are always PascalCase.

normalizeCase does transform identifiers for operations, data members, enumerators and constants.

Reply:
fixed